packages/fetchai/connections/p2p_libp2p/libp2p_node/aea/api.go

```diff
@@ -360,19 +359,31 @@ func UnmarshalEnvelope(buf []byte) (*Envelope, error) {
 	return envelope, err
 }
 
-func (aea *AeaApi) listen_for_envelopes() {
+func (aea *AeaApi) listenForEnvelopes() {
 	//TOFIX(LR) add an exit strategy
 	for {
-		envel, err := read_envelope(aea.pipe)
-		if err != nil {
-			logger.Error().Str("err", err.Error()).Msg("while receiving envelope")
+		envel, err := HandleAcnMessageFromPipe(aea.pipe, aea, aea.AeaAddress())
+
+		var e *common.PipeError
+
+		if errors.As(err, &e) {
+			logger.Error().
+				Str("err", err.Error()).
+				Msg("pip error while receiving envelope. disconnect")
 			logger.Info().Msg("disconnecting")
-			// TOFIX(LR) see above
 			if !aea.closing {
 				aea.stop()
```

Comment:
why this is not `Stop`? This function call will close the pipe. Once the pipe is closed, what is the job of `AeaApi` and the spawned coroutines?

Reply:
who will stop the other go coroutines, e.g. `envelopeSendLoop`?

Reply:
yeah, nice point.
two stop functions a bit confusing. now Stop will be used.